hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -325,6 +326,31 @@ public static SparkConf registerClasses(SparkConf conf) {
     }
   }
 
+  /**
+   * Deletes a bunch of keys from the Hoodie table, at the supplied commitTime
+   */
+  public JavaRDD<WriteStatus> delete(JavaRDD<HoodieKey> keys, final String commitTime) {
```

Comment:
With new write APIs coming in, I think it would make sense to record the action (upsert/insert/bulkinsert/delete) corresponding to the write batch in HoodieCommitMetadata itself. This could be a separate newbie task in itself.

Reply:
Can you open a ticket for this ?

Reply:
https://issues.apache.org/jira/browse/HUDI-331